File: src/main/java/by/dev/madhead/jarvis/util/AddressSearcher.java
package by.dev.madhead.jarvis.util;

import hudson.model.Cause.UserIdCause;
import hudson.model.Run;
import hudson.model.User;
import hudson.tasks.Mailer;

public class AddressSearcher {

    private AddressSearcher() {
    }

    /**
     * Finds email address of user with user id in Jenkins.
     *
     * @param userId user id in Jenkins
     * @return email address of user
     */
    public static String findById(String userId) {
        User user = User.getOrCreateByIdOrFullName(userId);
        return user.getProperty(Mailer.UserProperty.class).getAddress();
    }

    /**
     * Finds email address of user, who started run.
     *
     * @param run run
     * @return email address of user
     */
    public static String findBuilderAddress(Run<?, ?> run) {
        UserIdCause userIdCause = run.getCause(UserIdCause.class);
        return userIdCause != null ? findById(userIdCause.getUserId()) : null;
    }

}

File: src/main/kotlin/by/dev/madhead/jarvis/util/AddressSearcher.kt
package by.dev.madhead.jarvis.util

import hudson.model.Cause
import hudson.model.Run
import hudson.model.User
import hudson.tasks.Mailer

class AddressSearcher {

    fun findById(userId: String?): String? {
        return if (userId != null)
            User.getOrCreateByIdOrFullName(userId)
                    .getProperty(Mailer.UserProperty::class.java)
                    .address
        else null
    }

    fun findBuilderAddress(run: Run<*, *>): String? {
        return findById(run.getCause(Cause.UserIdCause::class.java)?.userId)
    }

}